catalog/to-consul/syncer.go

```diff
@@ -270,28 +290,28 @@ func (s *ConsulSyncer) watchService(ctx context.Context, name string) {
 // name that have the k8s tag and schedules them for removal.
 //
 // Precondition: lock must be held
-func (s *ConsulSyncer) scheduleReapServiceLocked(name string) error {
-	services, _, err := s.Client.Catalog().Service(name, s.ConsulK8STag, &api.QueryOptions{
-		AllowStale: true,
-	})
+func (s *ConsulSyncer) scheduleReapServiceLocked(name, namespace string) error {
+	// Set up query options
+	opts := api.QueryOptions{AllowStale: true}
+	if s.EnableNamespaces {
+		opts.Namespace = namespace
+	}
+
+	// Only consider services that are tagged from k8s
+	services, _, err := s.Client.Catalog().Service(name, s.ConsulK8STag, &opts)
 	if err != nil {
 		return err
 	}
 
+	// Create deregistrations for all of these
 	for _, svc := range services {
-		// If we have a namespace set and the key exactly matches this
-		// namespace, then we skip it.
-		if s.Namespace != "" &&
```

Comment:
again please explain why it's okay to delete

Reply:
from rebecca:
> with the addition of the allow and deny lists, this functionality has to change. before, it could be assumed that a sync pod was responsible for everything or only a single namespace. that is no longer the case, since now they are responsible for some arbitrary set of namespaces. because of this, the annotation no longer provides enough information to determine whether the sync pod should delete a specific service or not